content/docs/reference/routes/set-request-headers.mdx

```diff
@@ -18,20 +19,26 @@ import TabItem from '@theme/TabItem';
 
 ## Summary
 
-**Set Request Headers** allows you to set both static and dynamic values for given request headers. Static values can be useful if you want to pass along additional information to downstream applications as headers, or set authentication header to the request.
+**Set Request Headers** allows you to set both static and dynamic values for given request headers. Static values can be useful if you want to pass along additional information to upstream applications as headers, or to set a fixed authentication header on the request.
 
-The dynamic values enable you to pass ID and Access tokens in your requests to downstream applications.
+The dynamic values enable you to pass ID and Access tokens from your identity provider to upstream applications.
 
-See the configuration options below for examples of static and dynamic headers applied to a route.
+To pass dynamic values from the user's OIDC claim to an upstream service, see [JWT Claim Headers](../jwt-claim-headers).
+
+:::caution
+
+Neither HTTP/2 [pseudo-headers](https://www.rfc-editor.org/rfc/rfc9113.html#PseudoHeaderFields) (for example, `:authority`) nor the `Host:` header may be modified via this mechanism. Those headers may instead be modified via [`prefix_rewrite`](/docs/reference/routes/prefix-rewrite), [`regex_rewrite`](/docs/reference/routes/regex-rewrite), and [`host_rewrite`](/docs/reference/routes/host-rewrite).
+
+:::
 
 ## How to configure
 
 <Tabs>
 <TabItem value="Core" label="Core">
 
-| **YAML**/**JSON** setting | **Type** | **Usage**    |
-| :------------------------ | :------- | :----------- |
-| `set_request_headers`     | `string` | **optional** |
+| **YAML**/**JSON** setting | **Type**                          | **Usage**    |
+| :------------------------ | :-------------------------------- | :----------- |
+| `set_request_headers`     | `string` (map of key-value pairs) | **optional** |
```

Comment:
I still think it's a little confusing to start with `string` here. If you want to use "map of key-value pairs" I'd suggest removing the `string` part entirely.

I also wonder if the Kubernetes table **Type** column should be updated too? @wasaga: is the Kubernetes property also a map, or is it encoded as a plain string somehow?